adds documentation for the configuration options in local run (#558)
lightbend/cloudflow: docs/shared-content-source/docs/modules/develop/pages/cloudflow-local-sandbox.adoc

.`build.sbt`
[source, scala]
----
include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala/build.sbt[tag=local-conf]
...
----

==== Providing Configuration Values

With the local configuration file, you can provide values to streamlet configuration parameters and point volume mounts to locally available folders.

The configuration provided in this file follows the same principles as the general streamlet configuration system in Cloudflow: It uses HOCON syntax to define configuration entries, and it's hierarchically structured by streamlet under the `cloudflow.streamlets.<streamlet-name>` key.

For example, in the following configuration file, we give a local path to a `volume-mount` named `file-ingress` and provide a local value for a configuration parameter called `valid-logger`.

.Example of a `local.conf` file
[source, HOCON]
----
include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala/src/main/resources/local.conf[]
----

Note that because this file is in HOCON format, dot-notation is also supported.

[source, HOCON]
----

> cloudflow.streamlets.<streamlet-name>.<config-facet>.<config-key>=value

Reviewer: ```suggestion
cloudflow.streamlets.<streamlet-name>.<config-scope>.<config-key>=value
```

----

> === Supported Configuration Facets

Reviewer: ```suggestion
=== Supported Configuration Scopes
```

> In the sandbox runtime, only two configuration facets of streamlets are currently supported:

Reviewer: ```suggestion
In the sandbox runtime, only two configuration scopes of streamlets are currently supported:
```

> - configuration parameters, under the `config-parameters` facet.

Reviewer: facet -> section?

Author: yeah. I've been looking for a good term for that.
wdyt about _scope_?

Reviewer: hocon docs (once!) refers to config sections. I think config section is fine for now.

Reviewer: scope is good too.

> - configuration parameters, under the `config-parameters` facet.

Reviewer: ```suggestion
- configuration parameters, under the `config-parameters` scope.
```

> - volume mounts, under the `volume-mounts` facet.

Reviewer: ```suggestion
- volume mounts, under the `volume-mounts` scope.
```

The general syntax for a configuration entry is:

[source, HOCON]
----

> cloudflow.streamlets.valid-logger.<config-facet>.<config-key>=value

Reviewer: ```suggestion
cloudflow.streamlets.valid-logger.<config-scope>.<config-key>=value
```

----

> where the `config-facet` may be either `config-parameters` or `volume-mounts`.

Reviewer: ```suggestion
where the `config-scope` may be either `config-parameters` or `volume-mounts`.
```

> We can provide values for these two facets as we explain in the following section.

Reviewer: ```suggestion
We can provide values for these two scopes as we explain in the following section.
```

//...
----

Let’s assume that we have declared a `valid-logger` streamlet in the blueprint of this application. 
When running in the Sandbox, we can specify a custom value for this configuration parameter in the local configuration file as:

[source, HOCON]
----
include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala/src/main/resources/local-alt.conf[tag=config-param]
----

=== Using Volume Mounts

> In the Streamlet API, volume mounts --a Kubernets construct to access disk storage-- are declared in a similar way to configuration parameters, using a programmatic description of the volume mount that includes its desired mount path. 

Reviewer 2: ```suggestion
In the Streamlet API, volume mounts--a Kubernetes construct to access disk storage--are declared in a similar way to configuration parameters, using a programmatic description of the volume mount that includes its desired mount path.
```

In the following example, we declare a volume mount with name `source-data-mount`, requested to be mounted at `/mnt/data` in the pod's filesystem, and it requires to have a `ReadWriteMany` access mode:

[source,scala]
----
include::{cloudflow-examples-version}@docsnippets:ROOT:example$sensor-data-scala/src/main/scala/sensordata/SensorDataFileIngress.scala[tag=volume-mount1]
----

> In a Kubernetes deployment, that volume mount gets mapped to a permanent volume claim (PVC). 

Reviewer 2: ```suggestion
In a Kubernetes deployment, that volume mount gets mapped to a PersistentVolumeClaim (PVC). 
```
